processor/src/test/java/org/mapstruct/ap/test/targetthis/CustomerEntityMapper1.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.CollectionMappingStrategy;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(
+        unmappedSourcePolicy = ReportingPolicy.IGNORE,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED
+)
+public interface CustomerEntityMapper1 {
+    CustomerEntityMapper1 INSTANCE = Mappers.getMapper( CustomerEntityMapper1.class );
+
+    @Mapping(target = ".", source = "entity.*")
```

Comment:
I'm not sure about the `*` notation. Suppose that entity has another nesting level, would all elements from that level be mapped as well?
so, would both `prop1` and `prop2` be mapped?
`entity.prop1`
`entity.subentity.prop2`

Would this also happen with nested targets: so not `.` but `target = "target.prop2", source ="*"`?

Reply:
what would the latter mean if you have multiple source parameters? Like: `Target map(Source s1, Source2)`?

Reply:
> I'm not sure about the `*` notation. Suppose that entity has another nesting level, would all elements from that level be mapped as well?
> so, would both `prop1` and `prop2` be mapped?
> `entity.prop1`
> `entity.subentity.prop2`
> 
> Would this also happen with nested targets: so not `.` but `target = "target.prop2", source ="*"`?

In my head '*' means all attributes from that object, so 'entity.*' should try to map all properties from entity, 'entity.subentity.*' should map all properties from subentity, but not from entity.
'.' just way to say that mapping should be done to current target.
Ideally I would say that it would be good if it could target some other property as target 'target=entity.subentity source=dto.*', could try to map everything from 'dto' to 'entity.subentity', but it's not done in this PR.

Reply:
I'm missing some enters :smile: 

So, to be clear (just to make sure I interpret your text above correctly):
* `entity.*' should map all properties from entity (but **not** nested properties from `entity.subentity.`)
* `entity.subentity.*` should map all properties from `subentity` but **not** from `entity`..
??

Reply:
lets say there's source:

Source {
id;
created;
updated;
object;
}

Object {
prop1;
prop2;
}

@mapping("target=.", "source=Source.*")

should generate:

this.setId(source.id);
this.setCreated(source.created);
this.setUpdated(source.updated);
this.setObject(source.object); // or this.setObject( map(source.object) ) if there is special mapping for object.

Reply:
> I'm still analysing.. perhaps its good that @filiphr also voices his opinion..
> 
> However, I've the feeling that this PR solves 2 problems..
> 
> 1. #1406 (the target =".")
> 2. grouping of source properties:
> 
> Is 2. required to solve 1.? Otherwise it is perhaps better to split this into 2 PR's. The first one (this one) could solve the problem #1406... It requires testing some more scenarios.
> 
> The second one (the '*' notation to group properties) i'm still doubting about.. What does it offer extra..? We have name based mapping, so if names are the same in target and source property they are mapped automatically. You normally only have to specify the mappings that differ in name. (That can be switched off by an ignore option on bean mapping level: `BeanMapping#ignoreByDefault`)

1. Yes, it't related to #1406, as it allows to point to target object without actually writing name of that target.
2. to solve this, it requires some way to name, where multiple source properties should be mapped, so 1. is kinda required (or some other way, for example skipping target at all).

motivation here is that some parent objects in Source class hierarchy have fields that is always mapped to target objects, so to avoid repetition of copying same lines, we map all attributes to Target class, that have exactly same fields, just flat (aka in same level). This lets significantly decrease Mapper class (6 attributes in 50 classes, mapping both ways) and it reduces possible errors of missing to copy some of attributes and even worst if in future someone would new base class property that should be mapped to every single object.